### PR TITLE
feat： create

### PR DESCRIPTION
**Pull request by Numabeharuka**
# [Issue No.3](https://github.com/kenmori/handsonFrontend/blob/0087a459ae7e84af59ead2d97d9c6b5153481e69/work/markup/1.md#3)  
    <ul>
      <li><a href="a1.html">a1</a></li>
      <li><a href="a2.html">a2</a></li>
    </ul>

## [CodeSandBox](https://codesandbox.io/s/lesson3-lj8xde)
Latest commit: [f664f5d](https://github.com/Numabeharuka/Lesson-js/pull/4/commits/f664f5dc89140e568951ae1447332aecf5465d77)
### Anxious areas  
- The code is long, so if there is a better way to write it, please let me know.

### CODE REVIEW

**Comment by seeeen044** [comment]
Since similar items continue, how about iteration?

https://developer.mozilla.org/ja/docs/Web/JavaScript/Guide/Loops_and_iteration

File: lesson_03/script.js [hunk 1]
```diff
@@ -0,0 +1,12 @@
+"use strict";
+const ul = document.getElementById("js-target");
+for (let i = 1; i <= 2; i += 1) {
```
Comment (seeeen044) [inline lesson_03/script.js:3]: IMO
What is 2 of 2? It's a [magic number](https://ja.wikipedia.org/wiki/%E3%83%9E%E3%82%B8%E3%83%83%E3%82%AF%E3%83%8A%E3%83%B3%E3%83%90%E3%83%BC_(%E3%83%97%E3%83%AD%E3%82%B0%E3%83%A9%E3%83%A0)), so let's refrain from it.

File: lesson_03/script.js [hunk 2]
```diff
@@ -0,0 +1,12 @@
...
+  const li = document.createElement("li");
+  const a = document.createElement("a");
+  const img = document.createElement("img");
+  a.text = "a" + i;
+  a.href = "a" + i + ".html";
```
Comment (seeeen044) [inline lesson_03/script.js:8]:
```suggestion
  a.textContent = `a${i}`;
  a.href = `a${i}.html`;
```
IMO
You can write like this.
https://qiita.com/azukiazusa/items/b0d4a816142134ea7f34

**Review by hippopo-moo — COMMENTED** [review]
IMO
It is better to prepare the data first as follows, and then use it to build the DOM. 
That way, you can use the length of the data in the current magic number section as the number of times to turn the for statement.
How about having an array like this for values that change depending on the elements?

(This is an example)
```
const urls = [“a1.html”, “a2.html”]
```
(※データを見やすくurlだけの配列に変えました＆
次の課題と混ざったことを書いてしまっていました、、混乱させてごめんなさい🙇‍♀️
こちらで考えてみてもらえればと・・・🙏)

File: lesson_03/script.js [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+};
+for (let i = 1; i <= listCount(); i += 1) {
+  const li = document.createElement("li");
+  const a = document.createElement("a");
+  const img = document.createElement("img");
+  a.text = `a${i}`;
+  a.href = `${data[i - 1]}`;
+  img.src = "/img/bookmark.png";
+
+  ul.appendChild(li).appendChild(a).insertAdjacentElement("afterbegin", img);
```
Comment (seeeen044) [inline lesson_03/script.js:15]: Please try using `document.createDocumentFragment`.
If you use appendChild in a loop, the performance will be bad.
It's better to use and add that fragment at the end.

```
const fragment = document.createDocumentFragment();
{ // ループ内
fragment.appendChild(li).appendChild(a).insertBefore(img, a.firstChild);  
}
//最後にadd
ul.appendChild(fragment)

```

File: lesson_03/script.js [hunk 4]
```diff
@@ -0,0 +1,15 @@
+"use strict";
+const ul = document.getElementById("js-target");
+const data = ["a1.html", "a2.html"];
+const fragment = document.createDocumentFragment();
+for (let i = 1; i <= data.length; i += 1) {
+  const li = document.createElement("li");
+  const a = document.createElement("a");
+  const img = document.createElement("img");
+  a.text = `a${i}`;
+  a.href = `${data[i - 1]}`;
```
Comment (seeeen044) [inline lesson_03/script.js:10]:
```suggestion
  a.href = data[i-1];
```
IMO
There is no need to enclose it in a template literal, so it is easier to see if you remove it.

**Review by hippopo-moo — APPROVED** [review]

**Review by seeeen044 — APPROVED** [review]

**Pull request merged**